### ISSUE TITLE
2.1.0-rc1-final: No mapping to a relational type can be found for the CLR type 'ProductDto'.

### ISSUE THREAD

**Issue by VictorioBerra** [dotnet/efcore#12136]
Wrapping one of my assignments in a ternary statement causes the above exception.

## This code works
```csharp
context
.InventoryServer
.Select(x => new InventoryServerDto()
{
    Id = x.Id,
    Name = x.Name,
    InventoryServerProductLines = x.InventoryServerProductLines.Select(y => new InventoryServerProductLineDto()
    {
        InventoryServerId = y.InventoryServerId,
        ProductId = y.ProductId,
        Product =new ProductDto()
        {
            Id = y.Product.Id,
            Name = y.Product.Name
        }
    }).ToList()
}).ToList();
```

## This code throws the above exception
```csharp
context
.InventoryServer
.Select(x => new InventoryServerDto()
{
    Id = x.Id,
    Name = x.Name,
    InventoryServerProductLines = x.InventoryServerProductLines.Select(y => new InventoryServerProductLineDto()
    {
        InventoryServerId = y.InventoryServerId,
        ProductId = y.ProductId,
        Product = y.Product != null ? new ProductDto()
        {
            Id = y.Product.Id,
            Name = y.Product.Name
        } : null
    }).ToList()
}).ToList();
```

If you would like the see the full code, including DbContext and entities and DTOs they can be found in this sample repo:

https://github.com/VictorioBerra/AutoMapper.QueryableExtensions.Issue2643

### Further technical details
EF Core version: 2.1.0-rc1-final
Database Provider: Microsoft.EntityFrameworkCore.SqlServer
Operating system: Windows 10 64bit
IDE: Visual Studio 2017 15.7.0

**Comment by smitpatel** [comment]
Duplicate of #11937

**Comment by smitpatel** [comment]
Fixed in 2.1.0 RTM version.
